base/logallocator.go

```diff
@@ -55,10 +58,13 @@ func (alloc *LogAllocator) NewRecord(input []byte) (*LogRecord, string) {
 
 // Release releases this log record for recycling
 func (alloc *LogAllocator) Release(record *LogRecord) {
-	record._refCount--
+	atomic.AddInt32(&record._refCount, -1)
 	if record._refCount < 0 {
 		logger.Panic("negative reference count in record: ", record)
 	}
+	if record._refCount > 0 {
```

Comment:
both of this if at line 65 and the previous if at line 61 are wrong - since we call atomic to update the value (which assumes concurrent updates), it's the returned result that should be used for checking the value. e.g.

```go
val newRefCount = atomic.AddInt32(&record._refCount, -1)
if newRefCount < 0 {
...
```

BTW before fixing this, is there any lint analyzer that can detect this bug? That'd be very serious if the function has been called concurrently on the same record (there isn't in our situation).

Reply:
Yeah, I see it, but unfortunately the `atomic` analyzers in `govet` and `revive` linters don't seem to be able to spot this type of errors.
I suggest we remove atomic usage completely here, since I added it initially thinking that these will be used concurrently, which is not the case.